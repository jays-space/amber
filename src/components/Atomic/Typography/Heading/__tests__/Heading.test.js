import { screen, render } from "@testing-library/react";
import "@testing-library/jest-dom";

import { Heading } from "../";

describe("<Heading />", () => {
  const headingLabel = "amber software";

  it("should render correctly", () => {
    render(<Heading label={headingLabel} />);

    const headingElement = screen.getByRole("heading", { name: headingLabel });
    expect(headingElement).toBeInTheDocument();
    expect(headingElement).toMatchSnapshot();
  });

  it("should render capitalized label", () => {
    render(<Heading label={headingLabel} capitalize />);

    const headingElement = screen.getByRole("heading", { name: headingLabel });
    expect(headingElement).toBeInTheDocument();
  });

  describe("should render all header variants", () => {
    const variants = ["h1", "h2", "h3", "h4"];

    for (let variant of variants) {
      it(`should render ${variant} heading`, () => {
        render(<Heading variant={variant} label={`rendered ${variant} header`} />);

        const headingElement = screen.getByRole("heading", {
          name: `rendered ${variant} header`,
        });
        expect(headingElement).toBeInTheDocument();
      });
    }
  });
});
