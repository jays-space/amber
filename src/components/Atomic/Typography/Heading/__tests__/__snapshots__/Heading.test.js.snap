// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render correctly 1`] = `
<h1
  class=" text-slate-700 undefined text-4xl font-semibold"
>
  amber software
</h1>
`;
